home/tests/test_api.py

```diff
@@ -110,6 +110,32 @@ def test_tag_filtering(self, uclient):
         content = json.loads(response.content)
         assert content["count"] == 2
 
+    def test_search_aaq_filtering(self, uclient):
+        """
+        If an s filter is provided, only pages with matching search tearm are returned.
+        """
+        page1 = ContentPage.objects.first()
+        page1.enable_whatsapp = True
+        page1.save_revision().publish()
+        # it should return 1 page for correct search term
+        response = uclient.get("/api/v2/pages/?s=help&whatsapp=true")
+        content = json.loads(response.content)
+        assert content["count"] == 1
+        # it should return 0 pages for meaningless search term
+        response = uclient.get("/api/v2/pages/?s=#(&whatsapp=true")
+        content = json.loads(response.content)
+        assert content["count"] == 0
+        # it should return 0 pages for correct search term if no platform is provided
+        response = uclient.get("/api/v2/pages/?s=help")
+        content = json.loads(response.content)
+        print("content is ",content)
+        assert content["count"] == 0
+        # it should not return search term matching pages if they are unpublished
+        page1.unpublish()
+        response = uclient.get("/api/v2/pages/?s=#(&whatsapp=true")
```

Comment:
This is the same search term that previously didn't return any results. You probably want to use `s=help` here for the test to be valid

Reply:
Just to add to this, I don't think we should be using the `#` character in the URL for our tests for this. It's the fragment identifier character, and actually needs to be escaped if you want to use it in URL parameters, otherwise it could result in everything after the `#` being interpreted as a fragment identifier and not as part of the query string: https://www.rfc-editor.org/rfc/rfc3986.html#section-3.5